crates/bevy_ui/src/render/mod.rs

```diff
@@ -78,82 +78,94 @@ pub mod camera {
 }
 
 pub(crate) fn add_ui_graph(world: &mut World) {
-    let world = world.cell();
-    let mut graph = world.get_resource_mut::<RenderGraph>().unwrap();
-    let mut pipelines = world
-        .get_resource_mut::<Assets<PipelineDescriptor>>()
-        .unwrap();
-    let mut shaders = world.get_resource_mut::<Assets<Shader>>().unwrap();
-    let mut active_cameras = world.get_resource_mut::<ActiveCameras>().unwrap();
-    let msaa = world.get_resource::<Msaa>().unwrap();
+    world.resource_scope(|world, mut graph: Mut<RenderGraph>| {
```

Comment:
Deeply nested resource_scope's seems a bit suspicious. Is this driven by the other changes in this PR?

Reply:
This was the "quick and dirty" solution I called out in the description. The better fix would be to adapt the Render Graph to account for the System lifecycle, but thats more work than I wanted to invest in a proof of concept.